### ISSUE TITLE
struc.rdf doesn't work with custom box

### ISSUE THREAD

**Issue by entropybit** [biotite-dev/biotite#491]
The following code example where I try to caldulate the RDF with a custom box

```python
import biotite.structure as struc
import biotite.structure.io as strucio
import numpy as np

prot = strucio.load_structure("some_protein.pdb")
prot_ca = prot[prot.atom_name == "CA"]
r_com = struc.mass_center(prot_ca)
d=np.linalg.norm(prot_ca.coord - r_com.transpose(), axis=1)
#print(d.max())
box = np.array(
    [
        [2*d.max(), 0, 0],
        [0, 2*d.max(), 0],
        [0, 0, 2*d.max()]
    ]
)
prot_ca.coord = prot_ca.coord - r_com.transpose()
bins, g_r = struc.rdf(prot_ca, prot_ca, bins=10, box=box)
```

throws 
```
ValueError                                Traceback (most recent call last)
Cell In[1], line 18
     10 box = np.array(
     11     [
     12         [2*d.max(), 0, 0],
   (...)
     15     ]
     16 )
     17 prot_ca.coord = prot_ca.coord - r_com.transpose()
---> 18 bins, g_r = struc.rdf(prot_ca, prot_ca, bins=10, box=box)

File ~/miniconda3/envs/compbio/lib/python3.10/site-packages/biotite/structure/rdf.py:184, in rdf(center, atoms, selection, interval, bins, box, periodic)
    181     center = center.reshape((1,) + center.shape)
    183 if box.shape[0] != center.shape[0] or box.shape[0] != atom_coord.shape[0]:
--> 184     raise ValueError(
    185         "Center, box, and atoms must have the same model count"
    186     )
    188 # Calculate distance histogram
    189 edges = _calculate_edges(interval, bins)

ValueError: Center, box, and atoms must have the same model count
```

**Comment by JHKru** [comment]
Hi, your example works for me for with (1, 3, 3)-shaped boxes.
In rdf.py, input AtomArrays are seemingly always converted to AtomArrayStacks, but the shape of (3,3)-boxes passed as arguments to the function is seemingly not adjusted to match that.

Maybe 
```
    if box is None:
        if atoms.box is None:
            raise ValueError("A box must be supplied")
        else:
            box = atoms.box
```
could be changed to something like
```
    if box is None:
        if atoms.box is None:
            raise ValueError("A box must be supplied")
        else:
            box = atoms.box
    elif box.shape == (3, 3) and atoms.stack_depth() == 1:
        box = box[np.newaxis, :, :]
    else:
        raise ValueError("Boxes with dimension (m, 3, 3) have to be passed with AtomArrayStacks.")
```

**Comment by padix-key** [comment]
This seems like an elegant solution. I would merely replace `box.shape == (3, 3)` by `box.ndim == 2` to emphasize that this is about the number of dimensions. Would you like to open an PR?

**Comment by padix-key** [comment]
Fixed by #494
